test/inductor/test_torchinductor.py

```diff
@@ -7969,6 +7970,48 @@ def fn(x):
 
         self.common(fn, (torch.randn((16, 32)),), check_lowp=False)
 
+    @requires_cuda()
+    @torch._inductor.config.patch("layout_optimization", True)
+    @torch._inductor.config.patch("keep_output_stride", False)
+    @config.patch(implicit_fallbacks=True)
+    def test_custom_op_fixed_layout(self):
+        import torch.library
+
+        mod = nn.Conv2d(3, 128, 1, stride=1, bias=False).cuda()
+        inp = torch.rand(2, 3, 128, 128, device="cuda")
+        expected_stride = mod(inp).stride()
+
+        def foo_cpu(x):
+            self.assertEqual(x.stride(), expected_stride)
+            return x.clone()
+
+        def foo_cuda(x):
+            self.assertEqual(x.stride(), expected_stride)
+            return x.clone()
+
+        def foo_meta(x):
+            return torch.empty_like(x)
+
+        global libbar
+        if libbar is None:
+            libbar = torch.library.Library("bar", "DEF")
+            libbar.define(
+                "custom(Tensor self) -> Tensor", tags=[torch._C.Tag.needs_fixed_layout]
+            )
+            libbar.impl("custom", foo_cpu, "CPU")
+            libbar.impl("custom", foo_cuda, "CUDA")
+            libbar.impl("custom", foo_meta, "Meta")
+
+        def fn(x):
+            z = mod(x)
+            output = torch.ops.bar.custom(z)
+            return output
+
+        with torch.no_grad():
+            # With keep_output_stride False, inductor would normally have different layout from eager execution
+            # But because our custom op needs fixed layout, the assertions in the custom op will pass
+            self.common(fn, (inp,), check_lowp=False)
+
```

Comment:
Does this test fail without the `torch._C.Tag.needs_fixed_layout` tag?

Reply:
Yes it does!